### ISSUE TITLE
Remove symmetry Group initialization using strings

### ISSUE THREAD

**Issue by jaemolihm** [wannier-berri/wannier-berri#145]
Currently, it is possible to pass string to initialize a `Group`, e.g. `SYM.Group(["C4z"])`. I believe this feature gives little benefit (because one can just write `SYM.C4z`) and results in unnecessary complication. For example, `SYM.C3x` is not implemented. The error message for `SYM.Group(["C3x"], np.eye(3))` is very obscure while the error message for `SYM.Group([SYM.C3x], np.eye(3))` is clear.

```python3
>>> SYM.Group(["C3x"], np.eye(3))
Traceback (most recent call last):
  File "/home/jmlihm/program/wannier-berri/wannierberri/symmetry.py", line 194, in from_string_prod
    return product([globals()[s] for s in string.split("*")])
  File "/home/jmlihm/program/wannier-berri/wannierberri/symmetry.py", line 194, in <listcomp>
    return product([globals()[s] for s in string.split("*")])
KeyError: 'C3x'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jmlihm/program/wannier-berri/wannierberri/symmetry.py", line 220, in __init__
    sym_list=[(op if isinstance(op,Symmetry) else from_string_prod(op) )    for op in generator_list  ]
  File "/home/jmlihm/program/wannier-berri/wannierberri/symmetry.py", line 220, in <listcomp>
    sym_list=[(op if isinstance(op,Symmetry) else from_string_prod(op) )    for op in generator_list  ]
  File "/home/jmlihm/program/wannier-berri/wannierberri/symmetry.py", line 196, in from_string_prod
    raise ValueError("The symmetry {string} could not be recognuized : {}".format(err))
KeyError: 'string'

>>> SYM.Group([SYM.C3x], np.eye(3))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: module 'wannierberri.symmetry' has no attribute 'C3x'
```

Basically, one just needs to remove this line
https://github.com/wannier-berri/wannier-berri/blob/1fccb3564de3a12cacb6c1f5f5048614957de2cd/wannierberri/symmetry.py#L222
and adjust tests and examples.

**Comment by jaemolihm** [comment]
This was due to a bug in the error statement and got detected and fixed by flake8. Now the error message is fine.

```python3
>>> SYM.Group(["C3x"], np.eye(3))
Traceback (most recent call last):
  File "/home/jmlim/nlhe/GeTe/shift_and_injection/wannier-berri/wannierberri/symmetry.py", line 202, in from_string_prod
    return product([globals()[s] for s in string.split("*")])
  File "/home/jmlim/nlhe/GeTe/shift_and_injection/wannier-berri/wannierberri/symmetry.py", line 202, in <listcomp>
    return product([globals()[s] for s in string.split("*")])
KeyError: 'C3x'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/jmlim/nlhe/GeTe/shift_and_injection/wannier-berri/wannierberri/symmetry.py", line 230, in __init__
    sym_list = [(op if isinstance(op, Symmetry) else from_string_prod(op)) for op in generator_list]
  File "/home/jmlim/nlhe/GeTe/shift_and_injection/wannier-berri/wannierberri/symmetry.py", line 230, in <listcomp>
    sym_list = [(op if isinstance(op, Symmetry) else from_string_prod(op)) for op in generator_list]
  File "/home/jmlim/nlhe/GeTe/shift_and_injection/wannier-berri/wannierberri/symmetry.py", line 204, in from_string_prod
    raise ValueError(f"The symmetry {string} could not be recognized")
ValueError: The symmetry C3x could not be recognized
```
